File: PYTHON/1383 - Sudoku.py
def solve():
    ma = []
    v = [1,2,3,4,5,6,7,8,9]
    va = []
    flag = True
    
    for i in range(9):
        va = list(map(int,input().split()))
        ma.append(va)
        if sorted(va) != v:
            flag = False
            
    for i in range(9):
        va = []
        for j in range(9):
            va.append(ma[j][i])

        if sorted(va) != v:
            flag = False
            break
    
    if flag == False:
        return "NAO"
    else:

        flag = True
        for i in range(0,9,3):
            for j in range(0,9,3):
                va = []
                for x in range(i,i+3):
                    for y in range(j,j+3):
                        va.append(ma[x][y])
        
        
                va.sort()
                if va != v:
                    flag = False
                    break
                
            if flag == False:
                break
        
        if flag == False:
            return "NAO"
        else:
            return "SIM"
        
    
    
    
    


for w in range(int(input())):
    print('Instancia',w+1)
    ans = solve()
    print(ans)
    print()

File: C++/1383 - Sudoku.cpp
#include <bits/stdc++.h>
using namespace std;


void solve(){
    int i,j,x,y,flag =0;
    int ma[9][9];
    vector <int> va(9);

    vector <int> v= {1,2,3,4,5,6,7,8,9};

    for (i=0;i<9;i++){
        for(j=0;j<9;j++){
            cin >> ma[i][j];
        }
    }

    for (i=0;i<9;i++){
        vector <int> va(9);
        for (j=0;j<9;j++){
            va[j] = ma[i][j];
        }

        sort(va.begin(),va.end());
        if (va!=v){
            flag = 1;
            break;
        }
    }

    //cout<<flag<<endl;
    
    for (i=0;i<9;i++){
        vector <int> va(9);
        for (j=0;j<9;j++){
            va[j] = ma[j][i];
        }

        sort(va.begin(),va.end());
        if (va!=v){
            flag = 1;
            break;
        }
    }

    if (flag == 1){
        cout<<"NAO"<<endl;
    }

    else{
        for (i=0;i<9;i+=3){
            for (j=0;j<9;j+=3){
                vector <int> va;

                for (x=i;x<(i+3);x++){
                    for (y=j;y<(j+3);y++){
                        va.push_back(ma[x][y]);
                    }
                }
                sort(va.begin(),va.end());
                if (va!=v){
                    flag = 1;
                    break;
                }
                
            }

            if (flag == 1)break;

            
        }
    
        if (flag == 0)cout<<"SIM"<<endl;
        else cout<<"NAO"<<endl;

    }


}
int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
    int t;cin>>t;
    for (int p = 1;p<t+1;p++){
        cout<<"Instancia "<<p<<endl;
        solve();
        cout<<endl;

    }
    

}
